cmake/create_test.cmake

```diff
@@ -158,60 +155,72 @@ ginkgo_build_test_name(${test_name} test_target_name)
-    ginkgo_set_test_target_properties(${test_name} ${test_target_name})
-endfunction(ginkgo_create_hip_test)
+    ginkgo_set_test_target_properties(${test_target_name} ${ARGN})
+    ginkgo_add_test(${test_name} ${test_target_name} ${ARGN})
+endfunction(ginkgo_create_hip_test_internal)
 
 function(ginkgo_create_common_test test_name)
-    cmake_parse_arguments(PARSE_ARGV 1 common_test "" "" "DISABLE_EXECUTORS;ADDITIONAL_LIBRARIES")
-    set(executors)
     if(GINKGO_BUILD_OMP)
-        list(APPEND executors omp)
+        ginkgo_create_common_test_internal(${test_name} OmpExecutor omp ${ARGN})
     endif()
     if(GINKGO_BUILD_HIP)
-        list(APPEND executors hip)
+        ginkgo_create_common_test_internal(${test_name} HipExecutor hip ${ARGN})
     endif()
     if(GINKGO_BUILD_CUDA)
-        list(APPEND executors cuda)
+        ginkgo_create_common_test_internal(${test_name} CudaExecutor cuda ${ARGN})
     endif()
     if(GINKGO_BUILD_DPCPP)
-        list(APPEND executors dpcpp)
-    endif()
-    foreach(disabled_exec ${common_test_DISABLE_EXECUTORS})
-        list(REMOVE_ITEM executors ${disabled_exec})
-    endforeach()
-    foreach(exec ${executors})
-        ginkgo_build_test_name(${test_name} test_target_name)
-        # build executor typename out of shorthand
-        string(SUBSTRING ${exec} 0 1 exec_initial)
-        string(SUBSTRING ${exec} 1 -1 exec_tail)
-        string(TOUPPER ${exec_initial} exec_initial)
-        string(TOUPPER ${exec} exec_upper)
-        set(exec_type ${exec_initial}${exec_tail}Executor)
-        # set up actual test
-        set(test_target_name ${test_target_name}_${exec})
-        add_executable(${test_target_name} ${test_name}.cpp)
-        target_compile_features(${test_target_name} PUBLIC cxx_std_14)
-        target_compile_options(${test_target_name} PRIVATE ${GINKGO_COMPILER_FLAGS})
-        target_compile_definitions(${test_target_name} PRIVATE EXEC_TYPE=${exec_type} EXEC_NAMESPACE=${exec} GKO_COMPILING_${exec_upper})
-        target_link_libraries(${test_target_name} PRIVATE ${common_test_ADDITIONAL_LIBRARIES})
-        # use float for DPC++ if necessary
-        if((exec STREQUAL "dpcpp") AND GINKGO_DPCPP_SINGLE_MODE)
-            target_compile_definitions(${test_target_name} PRIVATE GINKGO_COMMON_SINGLE_MODE=1)
-            target_compile_definitions(${test_target_name} PRIVATE GINKGO_DPCPP_SINGLE_MODE=1)
-        endif()
-        ginkgo_set_test_target_properties(${test_name}_${exec} ${test_target_name})
-    endforeach()
+        ginkgo_create_common_test_internal(${test_name} DpcppExecutor dpcpp ${ARGN})
+    endif()
 endfunction(ginkgo_create_common_test)
 
-function(ginkgo_create_common_and_reference_test test_name)
-    ginkgo_create_common_test(${test_name})
+function(ginkgo_create_common_test_internal test_name exec_type exec)
+    cmake_parse_arguments(PARSE_ARGV 1 common_test "" "MPI_SIZE" "DISABLE_EXECUTORS;ADDITIONAL_LIBRARIES;ADDITIONAL_INCLUDES")
```

Comment:
Minor, should we store the list somewhere or have a macro which calls this? So that we only have one place to update the list if/when needed.